### PR TITLE
plugins/lspsaga: change backend plugin

### PR DESCRIPTION
**Pull request by GaetanLepage**
Since https://github.com/NixOS/nixpkgs/pull/247916, the `vimPlugins.lspsaga-nvim` is now fetching https://github.com/nvimdev/lspsaga.nvim.
The previously supported plugin is now unmaintained and this new one seems to be the most used version currently.
Hence, this PR changes the options of `plugins.lspsaga` accordingly.

I have taken care of marking all the removed options as such.
An explicit warning will provided to relevant users.

### CODE REVIEW

File: plugins/lsp/lspsaga.nix [hunk 1]
```diff
@@ -7,220 +7,651 @@
...
+          scrollUp = mkKeymapOption "<C-b>" "Scroll up.";
         };
-      };
 
-      maxDialogWidth = mkOption {
-        type = types.nullOr types.int;
-        default = null;
-        description = "Maximum dialog width";
-      };
+        requestTimeout = helpers.defaultNullOpts.mkInt 2000 "Request timeout.";
```
Comment (traxys) [inline plugins/lsp/lspsaga.nix:207]: This does not appear to be documented too

File: plugins/lsp/lspsaga.nix [hunk 2]
```diff
@@ -7,220 +7,651 @@
...
+            If it’s true, it will disable show the no response message.
+          '';
+
+          filter = helpers.defaultNullOpts.mkNullable (with types; attrsOf str) "{}" ''
+            Keys are LSP methods.
+            Values are a filter handler.
+            Function parameter are `client_id` and `result`.
+          '';
+
+          spInexist = helpers.defaultNullOpts.mkBool false "";
```
Comment (traxys) [inline plugins/lsp/lspsaga.nix:252]: This does not appear to be documented in https://nvimdev.github.io/lspsaga/finder/

File: plugins/lsp/lspsaga.nix [hunk 3]
```diff
@@ -7,220 +7,651 @@
...
 
-      extraConfigLua = ''
-        local saga = require 'lspsaga'
+    warnings =
+      mkIf (
+        # https://nvimdev.github.io/lspsaga/implement/#default-options
+        (isBool cfg.implement.enable && cfg.implement.enable)
+        && (isBool cfg.symbolInWinbar.enable && !cfg.symbolInWinbar.enable)
+      )
+      ["You have enabled the `implement` module but it requires `symbolInWinbar` to be enabled."];
```
Comment (traxys) [inline plugins/lsp/lspsaga.nix:462]: This is also required for `outline`
https://nvimdev.github.io/lspsaga/outline/

File: plugins/lsp/lspsaga.nix [hunk 4]
```diff
@@ -7,220 +7,651 @@
...
+
+          hideKeyword = helpers.defaultNullOpts.mkBool false "Hide keyword.";
+
+          showFile = helpers.defaultNullOpts.mkBool true "Show file.";
+
+          folderLevel = helpers.defaultNullOpts.mkInt 1 "Folder level.";
+
+          colorMode = helpers.defaultNullOpts.mkBool true "Color mode.";
+
+          dely = helpers.defaultNullOpts.mkInt 300 "Delay.";
```
Comment (traxys) [inline plugins/lsp/lspsaga.nix:350]: typo ? (delay)

**Review by traxys — CHANGES_REQUESTED** [review]

File: plugins/lsp/lspsaga.nix [hunk 5]
```diff
@@ -7,220 +7,651 @@
...
+            Show language server name.
+          '';
+
+          extendGitSigns = helpers.defaultNullOpts.mkBool false ''
+            Extend gitsigns plugin diff action.
+          '';
+
+          onlyInCursor = helpers.defaultNullOpts.mkBool true "";
+
+          maxHeight = helpers.defaultNullOpts.mkNullable (types.numbers.between 0.0 1.0) "0.3" "";
```
[the remark this answers is not in the document]
Comment (GaetanLepage) [reply plugins/lsp/lspsaga.nix:178]: Yes I know, but as they were used in the source code, I decided to put them in...

File: plugins/lsp/lspsaga.nix (hunk 3 above)
Comment (GaetanLepage) [reply plugins/lsp/lspsaga.nix]: Nice catch !

File: plugins/lsp/lspsaga.nix (hunk 4 above)
Comment (GaetanLepage) [reply plugins/lsp/lspsaga.nix:350]: Yes, but the option is mistyped upstream.
I will correct it in the module though.

File: plugins/lsp/lspsaga.nix (hunk 3 above)
Comment (GaetanLepage) [reply plugins/lsp/lspsaga.nix:462]: Well, as the `outline` module has no `enable` option, I don't think we need an additional warning here.

File: plugins/lsp/lspsaga.nix (hunk 5 above)
[the remark this answers is not in the document]
Comment (traxys) [reply plugins/lsp/lspsaga.nix:178]: I'd be more in favour of not including these options, as they could be internal or deprecated. If we really want we could clarify the situation with upstream.

File: plugins/lsp/lspsaga.nix [hunk 6]
```diff
@@ -7,220 +7,651 @@
...
-      renamePromptPrefix = mkOption {
-        type = types.nullOr types.str;
-        default = null;
-        description = "Rename prompt prefix";
-      };
-    };
-  };
+          sign = helpers.defaultNullOpts.mkBool true "Show sign in status column.";
+
+          lang = helpers.defaultNullOpts.mkNullable (with types; listOf str) "[]" ''
```
[the remark this answers is not in the document]
Comment (GaetanLepage) [reply plugins/lsp/lspsaga.nix:427]: removed

File: plugins/lsp/lspsaga.nix (hunk 2 above)
Comment (GaetanLepage) [reply plugins/lsp/lspsaga.nix:252]: removed

File: plugins/lsp/lspsaga.nix (hunk 1 above)
Comment (GaetanLepage) [reply plugins/lsp/lspsaga.nix:207]: removed

File: plugins/lsp/lspsaga.nix (hunk 5 above)
[the remark this answers is not in the document]
Comment (GaetanLepage) [reply plugins/lsp/lspsaga.nix:178]: removed

File: plugins/lsp/lspsaga.nix [hunk 7]
```diff
@@ -7,220 +7,651 @@
...
+
+          sign = helpers.defaultNullOpts.mkBool true "Show sign in status column.";
+
+          debounce = helpers.defaultNullOpts.mkInt 10 "Timer debounce.";
+
+          signPriority = helpers.defaultNullOpts.mkInt 40 "Sign priority.";
+
+          virtualText = helpers.defaultNullOpts.mkBool true "Show virtual text at the end of line.";
+
+          enableInInsert = helpers.defaultNullOpts.mkBool true "Enable in insert.";
```
[the remark this answers is not in the document]
Comment (GaetanLepage) [reply plugins/lsp/lspsaga.nix:198]: removed

File: plugins/lsp/lspsaga.nix [hunk 8]
```diff
@@ -7,220 +7,651 @@
...
-      package = helpers.mkPackageOption "lspsaga" pkgs.vimPlugins.lspsaga-nvim;
+        package = helpers.mkPackageOption "lspsaga" pkgs.vimPlugins.lspsaga-nvim;
 
-      signs = {
-        use = mkOption {
-          default = true;
-          type = types.bool;
-          description = "Whether to use diagnostic signs";
-        };
+        ui = {
```
[the remark this answers is not in the document]
Comment (GaetanLepage) [reply plugins/lsp/lspsaga.nix:54]: No, it should not be duplicated in each module.

File: plugins/lsp/lspsaga.nix [hunk 9]
```diff
@@ -7,220 +7,651 @@
...
+          priority = helpers.defaultNullOpts.mkInt 100 "Sign priority.";
         };
-      in
-        notEmpty keys;
 
-      code_action_keys = let
-        keys = {
-          quit = notNull cfg.keys.codeAction.quit;
-          exec = notNull cfg.keys.codeAction.exec;
+        beacon = {
```
[the remark this answers is not in the document]
Comment (GaetanLepage) [reply plugins/lsp/lspsaga.nix:424]: No, it should not be duplicated in each module.

File: plugins/lsp/lspsaga.nix [hunk 10]
```diff
@@ -7,220 +7,651 @@
...
+          virtualText = helpers.defaultNullOpts.mkBool true "Show virtual text at the end of line.";
+
+          enableInInsert = helpers.defaultNullOpts.mkBool true "Enable in insert.";
         };
 
-        info = mkOption {
-          type = types.nullOr types.str;
-          default = "  Info";
-          description = "Info diagnostic header";
+        scrollPreview = {
```
[the remark this answers is not in the document]
Comment (GaetanLepage) [reply plugins/lsp/lspsaga.nix:197]: No, it should not be duplicated in each module.

**Review by traxys — APPROVED** [review]

**Pull request merged**